homeassistant/components/ecobee/climate.py

```diff
@@ -647,6 +666,11 @@ def set_auto_temp_hold(self, heat_temp, cool_temp):
             self.hold_preference(),
             self.hold_hours(),
         )
+
+        # set_hold_temp was successful, reflect changes in thermostat
+        self.thermostat["runtime"]["desiredCool"] = cool_temp_setpoint * 10
+        self.thermostat["runtime"]["desiredHeat"] = heat_temp_setpoint * 10
```

Comment:
If you want to write the state optimistically, you'll need to call `self.schedule_update_ha_state()` or the entity won't be updated until the next poll

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L635

Reply:
Ah. Thanks. Taking a look at `schedule_update_ha_state()`. It looks like I'm going to need to work on the climate tests to support adding that call.